controllers/update_status.go

```diff
@@ -514,27 +502,20 @@ func validateProcessGroup(ctx context.Context, r *FoundationDBClusterReconciler,
 		return nil
 	}
 
-	_, idNum, err := podmanager.ParseProcessGroupID(processGroupStatus.ProcessGroupID)
-	if err != nil {
-		return err
-	}
-
-	specHash, err := internal.GetPodSpecHash(cluster, processGroupStatus.ProcessClass, idNum, nil)
+	correctPodSpec, err := internal.PodHasCorrectSpec(cluster, processGroupStatus.ProcessClass, processGroupStatus.ProcessGroupID, *pod)
 	if err != nil {
 		return err
 	}
 
-	incorrectPod := !metadataMatches(pod.ObjectMeta, internal.GetPodMetadata(cluster, processGroupStatus.ProcessClass, processGroupStatus.ProcessGroupID, specHash))
-	if !incorrectPod {
+	if correctPodSpec && !cluster.IsBeingUpgraded() {
```

Comment:
Is the implication of this that we will not set the `IncorrectPodSpec` condition until after the processes are running on the desired version? If so, is the reasoning that we will have a different condition set on the process groups when they're running at the wrong version? That seems like subtle logic, and it might be good to document.